balancer_conn_wrappers.go

```diff
@@ -429,14 +427,12 @@ func (acbw *acBalancerWrapper) getAddrConn() *addrConn {
 	return acbw.ac
 }
 
-var errSubConnNotReady = status.Error(codes.Unavailable, "SubConn not currently connected")
-
 // NewStream begins a streaming RPC on the addrConn.  If the addrConn is not
 // ready, returns errSubConnNotReady.
 func (acbw *acBalancerWrapper) NewStream(ctx context.Context, desc *StreamDesc, method string, opts ...CallOption) (ClientStream, error) {
-	transport := acbw.ac.getReadyTransport()
-	if transport == nil {
-		return nil, errSubConnNotReady
+	transport, err := acbw.ac.getTransport(ctx)
```

Comment:
Won't this cause a very long potential block on this NewStream() call? Or is this fine because you wrap this blocking operation with a context?

Reply:
Yes, it could, but it is fine because of the context.

Reply:
Optional: if it's easy, could you test this operation with a context with a defaultTestShortTimeout and make sure this NewStream() errors? I don't really feel strongly about this though, only if it'll be < 2ish mins to write.

Reply:
That was more like 10 minutes in the end, because it required debugging a hang (the select added will actually cause a hang if it hits because it pushes to the channel in a blocking way before connecting & returning a READY picker), and required me to realize I need to cancel the context manually, later, to make it not flaky.  Added.